x-pack/test/alerting_api_integration/spaces_only/tests/alerting/group1/create.ts

```diff
@@ -753,5 +753,115 @@ export default function createAlertTests({ getService }: FtrProviderContext) {
         });
       });
     });
+
+    describe('artifacts', () => {
+      describe('create rule with dashboards artifacts correctly', () => {
+        it('should not return dashboards artifacts in the rule response', async () => {
+          const response = await supertest
+            .post(`${getUrlPrefix(Spaces.space1.id)}/api/alerting/rule`)
+            .set('kbn-xsrf', 'foo')
+            .send(
+              getTestRuleData({
+                artifacts: {
+                  dashboards: [{ id: 'dashboard-1' }, { id: 'dashboard-2' }],
+                },
+              })
+            );
+          expect(response.status).to.eql(200);
+          objectRemover.add(Spaces.space1.id, response.body.id, 'rule', 'alerting');
+
+          expect(response.body.artifacts).to.be(undefined);
+        });
+
+        it('should store references correctly for dashboard artifacts', async () => {
+          const { body: createdDashboard } = await supertest
```

Comment:
nit: As we do not do any validation in the API if the dashboard exists, maybe we can save this extra call.

Reply:
Done in this [commit](https://github.com/elastic/kibana/pull/216292/commits/09260845c32fef31da7a5e4dc3d4c9a4e3932ae9)